api/base/middleware.py

```diff
@@ -339,7 +349,7 @@ def set_sloan_cookie(self, name: str, value, request, resp, custom_domain=None):
             resp.cookies[name]['domain'] = self.get_domain(request.environ['HTTP_REFERER'])
 
         if custom_domain:
-            resp.cookies[name]['domain'] = '.' + urlparse(custom_domain).netloc
+            resp.cookies[name]['domain'] = custom_domain
```

Comment:
This comment is more for L202/L205, but if someone, for some reason, set a preprint provider to have a custom domain but didn't set domain redirect, then this wouldn't end up being correct. L202/L205 should probably also check for `domain_redirect_enabled` and we should have a test where we check that case. While we're here.

Reply:
Also, is the `netloc` not still useful here? Or is our redirect code fragile enough that it really needs the custom domain to be specifically formatted (for example, I think it fails if it does or does not have a slash at the end)?

Reply:
Yeah I think your right on both these comments. I reverted the netloc stuff and added a check that the provider also have `domain_redirect_enabled` to get a custom cookie.